Library/Homebrew/utils/github/api.rb

```diff
@@ -256,6 +256,8 @@ def open_rest(url, data: nil, data_binary_path: nil, request_method: nil, scopes
     def paginate_rest(url, additional_query_params: nil, per_page: 100)
       (1..API_MAX_PAGES).each do |page|
         result = API.open_rest("#{url}?per_page=#{per_page}&page=#{page}&#{additional_query_params}")
+        break if result.blank?
```

Comment:
Could make this have one request fewer each time by using the link header to tell if you've just queried the last page: https://docs.github.com/en/rest/guides/using-pagination-in-the-rest-api?apiVersion=2022-11-28#using-link-headers

Reply:
Ship it now since it’s such a marked improvement and I’ll get to this later (after dinner)?

Reply:
I couldn't figure this out. I can see the link headers from the response normally, but the headers from the `curl` command used in `API.open_rest` are a string and parsing that seemed weird? I also couldn't find anywhere else that we paginate using these headers to get inspiration from.

Reply:
> headers from the `curl` command used in `API.open_rest` are a string and parsing that seemed weird

IMO parsing that (with regex) is reasonable to do here.